llvm/lib/Transforms/Vectorize/VPlanUnroll.cpp

```diff
@@ -295,8 +295,8 @@ void UnrollState::unrollRecipeByUF(VPRecipeBase &R) {
       continue;
     }
     if (auto *Red = dyn_cast<VPReductionRecipe>(&R)) {
-      auto *Phi = cast<VPReductionPHIRecipe>(R.getOperand(0));
-      if (Phi->isOrdered()) {
+      if (auto *Phi = dyn_cast<VPReductionPHIRecipe>(R.getOperand(0));
```

Comment:
nit: Perhaps this is personal preference, but I find it more readable to keep the variable out of the if statement so we avoid multi-statement lines, i.e. something like

```
  auto *Phi = dyn_cast<VPReductionPHIRecipe>(R.getOperand(0));
  if (Phi && Phi->isOrdered())
```

Reply:
Yeah that looks reasonable, I've no preference on this really. Will update

Reply:
Multiple statements as part of the condition are very uncommon in most parts of LLVM AFAICT, so for consistency it would be good to avoid.